### DISCUSSION TITLE
Chakra shorthand props not working

### DISCUSSION THREAD

**Discussion by DRD161** [chakra-ui/chakra-ui#9327] [Q&A] [answered]
I'm currently migrating to Chakra 3 and I noticed that none of the shorthand style props work. I am able to do: `paddingLeft="2"` but doing `pl="2"` gives me a type error: `TS2322: Type { children: Element | null; pl: string} is not assignable to type`. I have generated types using the CLI but those were for my theme and I am not sure why I would need to mess with Chakra's spacing props.

**Comment by segunadebayo** [comment]
Please double-check that when you create your theme, you did this

```ts
export const system = createSystem(defaultConfig, customConfig)
```

not

```ts
export const system = createSystem(customConfig)
```

**Comment by markodvornik** [comment]
I had the same TS error. However, it wasn't solved by correcting `createSystem(defaultConfig, customConfig)`.

It went away after modifying `tsconfig.json` with no specific setting, just modifying the file and restarting TS Server in VS Code. By restoring `tsconfig.json` to initial form the error hasn't return. It seems the error is not related to a specific setting, but it was a VS Code issue.

**Reply by markodvornik** [reply]
One more thing. Try deleting `node_modules`. 

I'm using `yarn` and `vite` and it seems that typescript can get in to strange state where basic props start to throw errors. Clearing caches (like `node_modules` and `.yarn`) seems to fix the problem.

packages:
```json
  "dependencies": {
    "@chakra-ui/react": "^3.13.0",
    "@emotion/react": "^11.14.0",
    "immer": "^10.1.1",
    "keycloakify": "^11.8.22",
    "next-themes": "^0.4.6",
    "react": "^19.0.0",
    "react-dom": "^19.0.0",
    "react-icons": "^5.5.0"
  },
  "devDependencies": {
    "@eslint/js": "^9.21.0",
    "@types/react": "^19.0.10",
    "@types/react-dom": "^19.0.4",
    "@vitejs/plugin-react": "^4.3.4",
    "eslint": "^9.21.0",
    "eslint-plugin-react-hooks": "^5.1.0",
    "eslint-plugin-react-refresh": "^0.4.19",
    "globals": "^15.15.0",
    "typescript": "~5.7.2",
    "typescript-eslint": "^8.24.1",
    "vite": "^6.2.0",
    "vite-tsconfig-paths": "^5.1.4"
  }
```

tsconfig:
```
{
  "compilerOptions": {
    "tsBuildInfoFile": "./node_modules/.tmp/tsconfig.app.tsbuildinfo",
    "target": "ES2020",
    "useDefineForClassFields": true,
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "module": "ESNext",
    "skipLibCheck": true,

    /* Bundler mode */
    "moduleResolution": "bundler",
    "allowImportingTsExtensions": true,
    "isolatedModules": true,
    "moduleDetection": "force",
    "noEmit": true,
    "jsx": "react-jsx",

    /* Linting */
    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true,
    "noUncheckedSideEffectImports": true
  },
  "include": ["src"]
}
```

**Comment by segunadebayo** [comment, marked as answer]
Shorthands come from `defaultConfig`. Call `createSystem(defaultConfig, yours)`.

If types still look stale, restart the TS server.
